docs/node/mainnet/e-upgrade.md

```diff
@@ -100,3 +100,309 @@ upgrade. For now, please take a look at the related [Testnet upgrade].
 :::
 
 [Testnet upgrade]: ../testnet/upgrade-log.md
+
+### Voting
+
+On XXX, an upgrade proposal will be proposed which (if accepted) will schedule
+the upgrade on epoch **XXX**.
+
+For an optimized voting experience, we recommend utilizing the [Oasis CLI].
+Follow these steps to cast your vote:
+1. [Import your keys into the wallet]
+2. [Cast your vote]
+
+### Upgrade Instructions
+
+The following steps should be performed only after the network has reached the
+upgrade epoch and has halted:
+
+1. Wait for the network to reach the upgrade epoch and halt. The node
+    will automatically stop without any action required on your part. After the
+    network has halted, proceed to the next steps.
+
+2. Download the Mainnet genesis file published in the
+   [Mainnet XXX release].
+
+:::info
+
+Mainnet state at epoch **XXX** will be exported and migrated to a 23.0.x
+compatible genesis file.
+
+The new genesis file will be published on the above link soon after reaching the
+upgrade epoch.
+
+:::
+
+3. Verify the provided Mainnet upgrade genesis file by comparing it to the
+   local network state dump.
+
+   Find the `genesis-mainnet-XXX.json` file in the `exports` subdirectory in
+   your data dir (e.g. `/node/`, `/srv/oasis/node/`)  and run `sha256sum` on it.
+   Afterwards, compare it with the hash that we will share on the
+   `#node-operators` Discord channel.
+
+   The state changes are described in the [State Changes](#state-changes)
+   section below.
+
+4. Replace the old genesis file with the new Mainnet genesis file.
+
+5. Ensure your node will remain stopped by disabling auto-starting via your
+   process manager (e.g., [systemd] or [Supervisor])
+
+6. Back up the entire data directory of your node. Verify that the backup
+includes the following folders:
+- for consensus: `tendermint/abci-state` and `tendermint/data`
+- for runtimes: `runtimes/*/mkvs_storage.badger.db` and
+`runtimes/*/worker-local-storage.badger.db`
+
+7. [Wipe state]. This must be performed _before_ replacing the Oasis Node
+   binary. In case you are upgrading ParaTimes/runtimes ensure you read the
+   following Danger section:
+
+:::danger
+
+State of ParaTimes/runtimes is not affected by this upgrade and MUST NOT be
+wiped. Wiping state for confidential ParaTimes will prevent your compute or
+key manager node from transitioning to the new network.
+
+To safely wipe the blockchain state on a runtime while preserving the runtime
+state, follow these steps:
+1. **Dry Run:** Initiate a dry run to preview which files will be deleted
+by running the following command: `oasis-node unsafe-reset --dry_run`
+2. **Wipe blockchain state:** After reviewing the dry run results, proceed with
+the reset by running: `oasis-node unsafe-reset`
+
+Note that by default, the `--preserve.mkvs_database` flag is set to true,
+preserving the runtime state. You can find additional information about the
+`unsafe-reset` on the [Wipe state] page.
```

Comment:
During the Testnet upgrade, we told a lot of operators to include this flag, so it might be confusing to them that now there is no flag. I do not know what the best option is. Do I remove this?